ccf_transaction_processor/Readme.md

```diff
@@ -6,37 +6,123 @@ https://creativecommons.org/licenses/by/4.0/
-"get_ledger_verifying_key" transaction.
-
-We note that the very first invocation of "get_ledger_verifying_key" rpc
-after the service is started is used to create the key pair, which will then be globally committed.
-The system admin who deploys the CCF ledger is expected to invoke the "get_ledger_verifying_key"
-rpc after deployment to generate these keys. Further, the admin must ensure that this key pair is globally committed
-before making the ledger available for PDO usage. This can checked by issuing the rpc again & ensuring that a verifying key gets returned. See https://microsoft.github.io/CCF/users/issue_commands.html for instructions on
-how an rpc can be issued via command line.
+The software is located under $PDO_SRC/transaction_processor/. The TP software is written and tested for CCF tag 0.9.2. Compatability with other CCF versions is not guaranteed.
+
+The instructions below may be followed to build and deploy CCF based PDO TP.
+
+# Get CCF Source Code
+
+CCF tag 0.9.2 is included as a submodule within PDO. Download the submodule using the following commands:
+
+cd $PDO_SRC
+git submodule init
+git submodule update
+
+
+# Install CCF Dependencies
+
+CCF/PDO combo has been tested under a scenario where CCF is deployed in a standalone VM. Further, the CCF/PDO combo
+has been tested only for the CCF virtual enclave mode. The dependencies needed to deploy CCF in an Ubuntu 18.04  VM
+with  virtual enclave mode can be installed by running the following command:
+
+$PDO_SRC/ccf_transaction_processor/CCF/getting_started/setup_vm/setup_nosgx.sh.
+
+The above script works only when the VM is not behind any proxies. If there are proxies, make the following changes
+before executing the above comamnd:
+
+        A. In the setup_nosgx.sh file, modify the command `sudo add-apt-repository ppa:ansible/ansible -y` by adding the `-E` option.
+           The modified command looks like `sudo -E add-apt-repository ppa:ansible/ansible -y`. It may also be necesssary to add a `sudo`
+           prefix to the `ansible-playbook ccf-dependencies-no-driver.yml` command appearing in the same file.
+
+        B. In ccf-dependencies-no-driver.yml and ccf-dependencies.yml files (present under the same folder), add the environment option for proxies
+           (see https://docs.ansible.com/ansible/latest/user_guide/playbooks_environment.html for reference)
+
+                environment:
+                    http_proxy: <specify-http-proxy-here>
+                    https_proxy: <specify-https-proxy-here>
+
+#Build CCF & PDO-TP
+
+Copy the transaction processor source files to the CCF src/apps/ folder
+
+mkdir $PDO_SRC/ccf_transaction_processor/CCF/src/apps/pdo_tp
+cp $PDO_SRC/ccf_transaction_processor/transaction_processor/*.* $PDO_SRC/ccf_transaction_processor/CCF/src/apps/pdo_tp/
```

Comment:
instead of copying, just adding a symlink might be easier?  In fact, couldn't you just add modify the paths in `add_ccf_app` to point to `../transaction_processor`?

Haven't tested either of the simplifications yet as i still struggle with ansible ...

Reply:
yes, thank you, fixed this!